### PR TITLE
use git submodules in the docker container

### PR DESCRIPTION
**Pull request by mbkumar**
This PR tries to fix the issue with simsopt import due to pybind11 related error.

### CODE REVIEW

**Comment by mbkumar** [comment]
This PR also changes the workflows by moving the CI to ubuntu-latest from ubuntu 18.04

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -52,7 +52,8 @@ jobs:
     - name: apt-get stuff needed for libstell and vmec
       run: |
         sudo apt-get update
-        sudo apt-get install -y build-essential gfortran openmpi-bin libopenmpi-dev libnetcdf-dev libnetcdff-dev liblapack-dev libblacs-mpi-dev libscalapack-mpi-dev libhdf5-dev libhdf5-serial-dev git m4 libfftw3-dev libboost-all-dev
+        # sudo apt-get install -y build-essential gfortran openmpi-bin libopenmpi-dev libnetcdf-dev libnetcdff-dev liblapack-dev libblacs-mpi-dev libscalapack-mpi-dev libhdf5-dev libhdf5-serial-dev git m4 libfftw3-dev libboost-all-dev
+        sudo apt-get install -y build-essential gfortran openmpi-bin libopenmpi-dev libnetcdf-dev libnetcdff-dev liblapack-dev libscalapack-mpi-dev libhdf5-dev libhdf5-serial-dev git m4 libfftw3-dev libboost-all-dev
```
Comment (florianwechsung) [inline .github/workflows/ci.yml:56]: Any reason to keep the commented out line in?

File: .github/workflows/extensive_ci.yml [hunk 2]
```diff
@@ -67,7 +67,8 @@ jobs:
     - name: apt-get stuff needed for libstell and vmec
       run: |
         sudo apt-get update
-        sudo apt-get install -y build-essential gfortran openmpi-bin libopenmpi-dev libnetcdf-dev libnetcdff-dev liblapack-dev libblacs-mpi-dev libscalapack-mpi-dev libhdf5-dev libhdf5-serial-dev git m4 libfftw3-dev
+        # sudo apt-get install -y build-essential gfortran openmpi-bin libopenmpi-dev libnetcdf-dev libnetcdff-dev liblapack-dev libblacs-mpi-dev libscalapack-mpi-dev libhdf5-dev libhdf5-serial-dev git m4 libfftw3-dev
+        sudo apt-get install -y build-essential gfortran openmpi-bin libopenmpi-dev libnetcdf-dev libnetcdff-dev liblapack-dev libscalapack-mpi-dev libhdf5-dev libhdf5-serial-dev git m4 libfftw3-dev
```
Comment (florianwechsung) [inline .github/workflows/extensive_ci.yml:71]: As above

**Review by florianwechsung — CHANGES_REQUESTED** [review]
What was the error message that you were seeing? Is this something that needs to be fixed outside of docker as well?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mbkumar) [reply .github/workflows/ci.yml:55]: > Any reason to keep the commented out line in?

No. Deleted the commented out lines.

**Review by florianwechsung — APPROVED** [review]

**Pull request merged**